x-pack/plugins/reporting/server/config/ui_settings.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, UiSettingsParams } from 'kibana/server';
+import { PLUGIN_ID, UI_SETTINGS_CUSTOM_PDF_LOGO } from '../../common/constants';
+
+const kbToBase64Length = (kb: number) => Math.floor((kb * 1024 * 8) / 6);
+
+// inspired by x-pack/plugins/canvas/common/lib/dataurl.ts
+const dataurlRegex = /^data:([a-z]+\/[a-z0-9-+.]+)(;[a-z-]+=[a-z0-9-]+)?(;([a-z0-9]+))?,/;
+const imageTypes = ['image/svg+xml', 'image/jpeg', 'image/png', 'image/gif'];
+
+export const isImageData = (str: any): boolean => {
+  if (typeof str !== 'string') {
+    return false;
+  }
+
+  const matches = str.match(dataurlRegex);
+
+  if (!matches) {
+    return false;
+  }
+
+  const [, mimetype, , , encoding] = matches;
+  const imageTypeIndex = imageTypes.indexOf(mimetype);
+  if (imageTypeIndex < 0 || encoding !== 'base64') {
+    return false;
+  }
+
+  return true;
+};
+
+export function registerUiSettings(core: CoreSetup<object, unknown>) {
+  core.uiSettings.register({
+    [UI_SETTINGS_CUSTOM_PDF_LOGO]: {
+      name: i18n.translate('xpack.reporting.pdfFooterImageLabel', {
+        defaultMessage: 'PDF footer image',
+      }),
+      value: null,
+      description: i18n.translate('xpack.reporting.pdfFooterImageDescription', {
+        defaultMessage: `Custom image to use in the PDF's footer`,
+      }),
+      sensitive: true,
+      type: 'image',
+      schema: schema.nullable(
+        schema.any({
```

Comment:
The [original schema](https://github.com/elastic/kibana/pull/94746/files#diff-4ccb97b5b86ca2e9c73c83ee67cda58ea7bc314f2cde72da2e05527ffd4736a9L58) defined a max byte size, and it looks like we lost that here. Can we add this validation back into the schema?

Reply:
Definitely! 1d76ec9a2fc